kv/dist_sender_test.go

```diff
@@ -568,3 +574,64 @@ func TestVerifyPermissions(t *testing.T) {
 	}
 	n.Stop()
 }
+
+// TestSendRPCRetry verifies that sendRPC failed on first address but succeed on
+// second address, the second reply should be successfully returned back.
+func TestSendRPCRetry(t *testing.T) {
+	g := makeTestGossip(t)
+	if err := g.SetNodeDescriptor(&proto.NodeDescriptor{NodeID: 1}); err != nil {
+		t.Fatal(err)
+	}
+	// Fill RangeDescriptor with 2 replicas
+	var descriptor = proto.RangeDescriptor{
+		RaftID:   1,
+		StartKey: proto.Key("a"),
+		EndKey:   proto.Key("z"),
+	}
+	for i := int32(1); i <= 2; i++ {
+		addr := util.MakeUnresolvedAddr("tcp", fmt.Sprintf("node%d", i))
+		nd := &proto.NodeDescriptor{
+			NodeID: proto.NodeID(i),
+			Address: proto.Addr{
+				Network: addr.Network(),
+				Address: addr.String(),
+			},
+		}
+		if err := g.AddInfo(gossip.MakeNodeIDKey(proto.NodeID(i)), nd, time.Hour); err != nil {
+			t.Fatal(err)
+		}
+
+		descriptor.Replicas = append(descriptor.Replicas, proto.Replica{
+			NodeID:  proto.NodeID(i),
+			StoreID: proto.StoreID(i),
+		})
+	}
+	// Define our rpcSend stub which return second address trying value.
```

Comment:
`s/which return second address trying value./which returns success on the second address./g`